tests/rule_tests/remove_if_expression.rs

```diff
@@ -0,0 +1,23 @@
+use darklua_core::rules::{RemoveIfExpression, Rule};
+
+test_rule!(
+    remove_if_expression,
+    RemoveIfExpression::default(),
+    assign_if_expression("local a = if true then 1 else 2") => "local a = (function() if true then return 1 else return 2 end end)()",
+	assign_if_expression_with_elseif("local a = if true then 1 elseif false then 2 else 3") => "local a = (function() if true then return 1 elseif false then return 2 else return 3 end end)()"
+);
```

Comment:
Let's add more tests! For example, we need to treat function calls and `...` in a specific way to make sure we are not returning multiple values. For that we need to wrap them in parentheses.